File: VB/Background Services/RandomLoggingVB.Service/Extensions/ServicesExtension.vb
Imports System.Runtime.CompilerServices
Imports Microsoft.Extensions.DependencyInjection
Imports Microsoft.Extensions.Hosting

Public Module ServicesExtension

    <Extension>
    Public Function AddRandomBackgroundService(builder As HostApplicationBuilder) As HostApplicationBuilder

        builder.Services.AddSingleton(Of RandomLoggingService)
        builder.Services.AddHostedService(Function(service) service.GetRequiredService(Of RandomLoggingService))

        Return builder

    End Function

End Module

File: CSharp/Applications/WpfLoggingAttrDI/Extensions/ServicesExtension.cs
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;

namespace WpfLoggingAttrDI.Service;

public static class ServicesExtension
{
    public static HostApplicationBuilder AddRandomBackgroundService(this HostApplicationBuilder builder)
    {
        builder.Services.AddSingleton<RandomLoggingService>();
        builder.Services.AddHostedService(service => service.GetRequiredService<RandomLoggingService>());

        return builder;
    }
}